Week3/homework/7-step3.js

```diff
@@ -22,4 +22,9 @@ f2(y);
 console.log(y);
 
 // Add your explanation as a comment here
+/*In Pass by Value, Function is called by directly passing the value of the variable as the argument. Changing the argument inside the function doesn’t affect the variable passed from outside the function.
+Javascript always pass by value so changing the value of the variable never changes the underlying primitive (String or number).
+However, when a variable refers to an object which includes array, the value is the reference to the object.
+In Pass by Reference, Function is called by directly passing the reference/address of the variable as the argument. Changing the argument inside the function affect the variable passed from outside the function. In Javascript objects and arrays follows pass by reference.
+so if we are passing object or array as an argument to the method, then  that value of the object will change.*/
```

Comment:
This looks like a straight copy & paste from https://codeburst.io/javascript-pass-by-value-and-pass-by-reference-in-javascript-fcf10305aa9c

Do you actually understand what is happening?